internal/pkg/mapper/defaultbucket/local_save.go

```diff
@@ -0,0 +1,50 @@
+package defaultbucket
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/orderedmap"
+)
+
+// MapBeforeLocalSave - replace default buckets in IM with placeholders.
+func (m *defaultBucketMapper) MapBeforeLocalSave(recipe *model.LocalSaveRecipe) error {
+	config, ok := recipe.Object.(*model.Config)
+	if !ok {
+		return nil
+	}
+
+	return m.visitStorageInputTables(config, m.replaceDefaultBucketWithPlaceholder)
+}
+
+func (m *defaultBucketMapper) replaceDefaultBucketWithPlaceholder(config *model.Config, sourceTableId string, inputTable *orderedmap.OrderedMap) error {
+	sourceConfigPath, found := m.getDefaultBucketSourceConfigurationPath(config, sourceTableId)
+	if !found {
+		return nil
+	}
+
+	tableName := strings.Split(sourceTableId, ".")[2]
+	inputTable.Set(`source`, fmt.Sprintf(`{{:default-bucket:%s}}.%s`, sourceConfigPath, tableName))
+
+	return nil
+}
+
+func (m *defaultBucketMapper) getDefaultBucketSourceConfigurationPath(config *model.Config, tableId string) (string, bool) {
+	componentId, configId, match := m.State.Components().MatchDefaultBucketInTableId(tableId)
+	if !match {
+		return "", false
+	}
+
+	sourceConfigKey := model.ConfigKey{
+		BranchId:    config.BranchId,
+		ComponentId: componentId,
+		Id:          configId,
+	}
+	sourceConfig, found := m.State.Get(sourceConfigKey)
+	if !found {
+		m.Logger.Warnf(`Warning: configuration "%s" of component "%s" that was supposed to create table "%s" in the input mapping of configuration "%s" not found`, configId, componentId, tableId, config.Id)
```

Comment:
Bolo by lepsie vracat `error` a az na najvyssej urovni vypisat vsetko ako warning:
https://github.com/keboola/keboola-as-code/blob/ef7e1e8ab2a7e01731bcadd4f6b33bb780f9b120/internal/pkg/mapper/orchestrator/remote_load.go#L37-L40

Jedna bude iba 1x `Warning`, a chyby budu na dalsich riadkoch s `- ....`.
A ak bude potrebne zmenit logiku a spravit z toho chyby, tak to bude jednoduchsie.

Reply:
+ ta chybova hlaska mi pride dost dlha, rozdelil by som ju nejak takto:
https://github.com/keboola/keboola-as-code/blob/74ebd25f590aa184c2b33487cae7344709b0ab91/internal/pkg/mapper/sharedcode/links/local_load.go#L41-L46

Napr.

```
%s not found
- referenced from input mapping
- from %s
```

Reply:
`configuration "%s" of component "%s"`  na to mame metodu `Desc()`.
Je ju mozne zavolat na lubovolnym `key` a vrati napr. `config "branch:456/component:keboola.foo/config:234"`
A je ju tiez mozne zavolat nad `ObjectManifest` a vrati `config "branch/other/orchestrator"`.

Teda zavisi od toho co chces vypisat. 
V `remote load` by som pouzil 1. sposob s ID.
V ostatnych pripadoch by som pouzival `ObjectManifest.Desc`, kedze vypise `path`  a user vie kde hladat problem.

V nejakom starsom kode to takto ^^^^ este nemusi byt spravene.

Reply:
Ty warningy se pak ale můžou potenciálně zamíchat s chybějící branchí: https://github.com/keboola/keboola-as-code/blob/dc161bd0d7195beed271ed82de5a73e9f306f4d5/internal/pkg/mapper/defaultbucket/local_load.go#L33 co by potom ale neskončilo jako error ale jen warning. To neva?

Reply:
Jo, díky, ty hlášky upravím. Moc jsem nevěděl co tam vlastně napsat. 😃

Reply:
tu branch by som vyriesil tak,  ze by som tam dal `MustGet`, ... ak je nacitany nejaky objekt ale nie je jeho branch, tak je nieco definitivne zle a nema zmysel pokracovat.

Reply:
`remote/local` load nacitava najprv `branch` a az potom objekty z nej.